src/api/commands/address/AddressAddCommand.ts

```diff
@@ -85,14 +86,74 @@ export class AddressAddCommand extends BaseCommand implements RpcCommandInterfac
         return await this.addressService.create(newAddress);
     }
 
+
     public async validate(data: RpcRequest): Promise<RpcRequest> {
-        //
+        if (data.params.length < 1) {
+            throw new MissingParamException('profileId');
+        }
+        if (typeof data.params[0] !== 'number') {
+            throw new InvalidParamException('profileId');
+        }
+        if (data.params.length < 2) {
+            throw new MissingParamException('title');
+        }
+        if (typeof data.params[1] !== 'string') {
+            throw new InvalidParamException('title');
+        }
+        if (data.params.length < 3) {
+            throw new MissingParamException('firstName');
+        }
+        if (typeof data.params[2] !== 'string') {
+            throw new InvalidParamException('firstName');
+        }
+        if (data.params.length < 4) {
+            throw new MissingParamException('lastName');
+        }
+        if (typeof data.params[3] !== 'string') {
+            throw new InvalidParamException('lastName');
+        }
+        if (data.params.length < 5) {
+            throw new MissingParamException('addressLine1');
+        }
+        if (typeof data.params[4] !== 'string') {
+            throw new InvalidParamException('addressLine1');
+        }
+        if (data.params.length < 6) {
+            throw new MissingParamException('addressLine2');
+        }
+        if (typeof data.params[5] !== 'string') {
+            throw new InvalidParamException('addressLine2');
+        }
+        if (data.params.length < 7) {
+            throw new MissingParamException('city');
+        }
+        if (typeof data.params[6] !== 'string') {
+            throw new InvalidParamException('city');
+        }
+        if (data.params.length < 8) {
+            throw new MissingParamException('state');
+        }
+        if (typeof data.params[7] !== 'string' && data.params[7] !== '') {
+            throw new InvalidParamException('state');
+        }
+        if (data.params.length < 9) {
+            throw new MissingParamException('country');
+        }
+        if (typeof data.params[8] !== 'string') {
+            throw new InvalidParamException('country');
+        }
+        if (data.params.length < 10) {
+            throw new MissingParamException('zipCode');
+        }
+        if (typeof data.params[9] !== 'string') {
+            throw new InvalidParamException('zipCode');
+        }
         return data;
```

Comment:
if type exists, it should be AddressType

Reply:
typeof doesn't support AddressType

Reply:
const type = data.params[x];
const validTypes = [AddressType.X, AddressType.Y, AddressType.Z];
if (type && !_.includes(validTypes, type)) {
    throw new Exception();
...